src/main/java/org/springframework/data/elasticsearch/repository/query/ElasticsearchStringQuery.java

```diff
@@ -71,17 +72,23 @@ public Object execute(Object[] parameters) {
 		StringQuery stringQuery = createQuery(accessor);
```

Comment:
In a later stage, we should move the query execution into a strategy object (see [`MongoQueryExecution`](https://github.com/spring-projects/spring-data-mongodb/blob/master/spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/query/MongoQueryExecution.java#L53) to execute the query according to what we want to see as results (see also [MongoDB](https://github.com/spring-projects/spring-data-mongodb/blob/master/spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/query/AbstractMongoQuery.java#L129-L156)).

Unconditionally calling `unwrapSearchHits` seems a workaround for a missing execution model.

Reply:
Just a short reply before I find more time:

- I think the aggregations are up to now contained in the `AggregatedPage` because that was the only possibility to return additional information for a search result. From the ES concept, aggregations are an addition to a search result, so they should be moved to the `SearchHits` class (and I would not keep the `AggregatedPage`.
- Renaming from `query..()` to `search..()`: yes, but I would not mix this with this PR but do a follow up on this.
- I'll check the `scroll` methods and see how adding `SearchSlice` makes this cleaner, probably we should separate the scrolling from the paging.
- Totally replacing page with scroll would probably make sense as ES doesn't have the concept of pages and scrolling is faster than searching with offset and size, but that would have breaking changes for the repository interface as well, as up to now repositories can return pages and not the scrollid, and have to check if the scrollid is supported in the `Query`implementations.

Up to now the breaking changes are in the `SearchOperations` implementation with the repositories working as before; I'd like to keep that this way for now.

Reply:
What do you think about reducing this PR to introducing the new `SearchHit`/`SearchHits`/… types only? I agree that the method changes are a better fit for a separate PR.

Reply:
that would be commit f4203f9 plus adding some info to the `SearchHits` plus some more info to the class. I'll get that running and make further tickets for changing the methods, moving aggregates from the page to SearchHits etc.

Reply:
>Totally replacing page with scroll would probably make sense as ES doesn't have the concept of pages and scrolling is faster than searching with offset and size, but that would have breaking changes for the repository interface as well, as up to now repositories can return pages and not the scrollid, and have to check if the scrollid is supported in the Query implementations.

Scrolling is a stateful operation in ES, while searching is stateless. Using Scroll as the default for repo methods that return a page is not a good idea.

Reply:
> Totally replacing page with scroll would probably make sense

I was referring to `startScroll` and `continueScroll` methods which return a `Page` object. Both should rather return a `Slice` type. Repository methods are not affected by this change.

Reply:
Ok sorry for the noise here.